app/src/main/java/com/duckduckgo/app/global/shortcut/AppShortcutCreator.kt

```diff
@@ -65,28 +68,29 @@ class AppShortcutCreatorLifecycleObserver(
     @UiThread
     override fun onCreate(owner: LifecycleOwner) {
         Timber.i("Configure app shortcuts")
-        appShortcutCreator.configureAppShortcuts()
+        appShortcutCreator.refreshAppShortcuts()
     }
 }
 
 @SingleInstanceIn(AppScope::class)
+@ContributesBinding(AppScope::class)
 class AppShortcutCreator @Inject constructor(
     private val context: Context,
     @AppCoroutineScope private val appCoroutineScope: CoroutineScope,
     private val appBuildConfig: AppBuildConfig,
-    private val duckChat: DuckChat,
+    private val duckChatProvider: Provider<DuckChat>,
     private val dispatchers: DispatcherProvider,
-) {
+) : AppShortcuts {
 
-    fun configureAppShortcuts() {
+    override fun refreshAppShortcuts() {
         appCoroutineScope.launch(dispatchers.io()) {
             val shortcutList = mutableListOf<ShortcutInfo>()
 
             shortcutList.add(buildNewTabShortcut(context))
             shortcutList.add(buildClearDataShortcut(context))
             shortcutList.add(buildBookmarksShortcut(context))
 
-            if (duckChat.isEnabled()) {
+            if (duckChatProvider.get().showInBrowserMenu()) {
```

Comment:
What if we made the `showInBrowserMenu` a state flow instead of a boolean getter (or add an alternative function to the interface with a state flow)? Then we would avoid the cyclic dependency between `DuckChat` and `AppShortcuts`, and we wouldn't need the `Provider` workaround.

Reply:
I've made it a `StateFlow` in https://github.com/duckduckgo/Android/pull/6002 👍

Reply:
Nice! I only see `showInAddressBar` as state flow in https://github.com/duckduckgo/Android/pull/6002 though, would you mind doing the same for `showInBrowserMenu` here?

Reply:
Oh sorry, my mistake 😂 Yeah, I'll update it here as well.